topic/machine-learning/llm-langchain/requirements.txt

```diff
@@ -5,20 +5,22 @@ cratedb-toolkit==0.0.12
 
 # langchain[cratedb,openai]==0.0.354
 # langchain-community==0.0.8
-langchain-openai==0.0.6
+
+google-cloud-aiplatform<2
+langchain-google-vertexai<2
+langchain-openai<0.2
+langchain-text-splitters<0.3
 pueblo[cli,nlp]>=0.0.7
 pydantic>=1,<3
 pypdf<5
 python-dotenv<2
 requests<3
 requests-cache<2
 sqlalchemy==2.*
-unstructured<0.12
-google-cloud-aiplatform
-langchain-google-vertexai
+unstructured<0.15
 
 # Development.
 # cratedb-toolkit @ git+https://github.com/crate-workbench/cratedb-toolkit.git@main
-langchain[cratedb,openai] @ git+https://github.com/crate-workbench/langchain.git@cratedb#subdirectory=libs/langchain
-langchain-community @ git+https://github.com/crate-workbench/langchain.git@cratedb#subdirectory=libs/community
+langchain[cratedb,openai] @ git+https://github.com/crate-workbench/langchain.git@cratedb-next#subdirectory=libs/langchain
+langchain-community @ git+https://github.com/crate-workbench/langchain.git@cratedb-next#subdirectory=libs/community
```

Comment:
> For validation purposes, 9d8ffc160c8 temporarily pulls in upstream adjustments for LangChain 0.2 from the staging branch cratedb-next.

This commit needs to be removed before merging, and after updating `cratedb` with ingredients from `cratedb-next` over at https://github.com/crate-workbench/langchain.

Reply:
The commit has been removed after merging the update adjustments on our fork.